core/unit_test/TestUtilities.hpp

```diff
@@ -371,4 +371,22 @@ inline void test_utilities() {
   }
 }
 
+template <std::size_t... Idxs, class... Args>
+std::size_t do_comma_emulation_test(std::integer_sequence<std::size_t, Idxs...>,
+                                    Args... args) {
+  // Count the bugs, since ASSERT_EQ is a statement and not an expression
+  std::size_t bugs = 0;
+  // Ensure in-order evaluation
+  std::size_t i = 0;
+  KOKKOS_IMPL_FOLD_COMMA_OPERATOR(bugs += std::size_t(Idxs != i++) /*, ...*/);
+  // Ensure expansion of multiple packs works
+  KOKKOS_IMPL_FOLD_COMMA_OPERATOR(bugs += std::size_t(Idxs != args) /*, ...*/);
```

Comment:
I am not suggesting to change, I am writing down what it would take to write without the macro.
```suggestion
  (void)std::initializer_list<int>{ (bugs += std::size_t(Idxs != i++), 0)... };
  (void)std::initializer_list<int>{ (bugs += std::size_t(Idxs != args), 0)... };
```